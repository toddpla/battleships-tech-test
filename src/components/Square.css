.square {
  border: 1px solid black;
  height: 20px;
  width: 20px;
  text-align: center;
}



.square:hover {
  color: white;
  background-color: black;
  cursor: pointer;
}

.square-button{
   display:inline-block;
   width: 20px;
   height: 20px;
}

.square-selected {
  color: white;
  background-color: black;
}

.square-ship {
  background-color: black;
}
;
