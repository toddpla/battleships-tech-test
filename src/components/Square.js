import React from 'react'
import './Square.css'

export class Square extends React.Component {

  handleSelect = () => {this.props.handleSelect(this.props.id)}

  render() {
    return (
      <td
        id={`square-${this.props.id}`}
        className={`square ${this.props.selected ? "square-selected" : "square-unselected"} square-${this.props.condition}`}
        onClick={this.handleSelect}
        >
      </td>
    )
  }
}

export default Square;
