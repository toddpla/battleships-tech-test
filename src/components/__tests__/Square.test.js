import React from 'react'
import { Square } from '../Square'
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const square = renderer.create(<Square />).toJSON();
  expect(square).toMatchSnapshot();
});
