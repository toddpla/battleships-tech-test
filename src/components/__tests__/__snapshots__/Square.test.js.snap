// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<td
  className="square square-unselected square-undefined"
  disabled={true}
  id="square-undefined"
  onClick={[Function]}
/>
`;
